OnlineShop/OnlineShop.BLL/Services/Implementations/UserService.cs

```diff
@@ -1,8 +1,73 @@
-﻿using OnlineShop.BLL.Services.Interfaces;
+﻿using FluentValidation;
+using Mapster;
+using OnlineShop.BLL.DTO.Request;
+using OnlineShop.BLL.DTO.Response;
+using OnlineShop.BLL.Exceptions;
+using OnlineShop.BLL.Helpers;
+using OnlineShop.BLL.Mappers;
+using OnlineShop.BLL.Services.Interfaces;
+using OnlineShop.BLL.Validators;
+using OnlineShop.DAL.Entities.Implementations;
+using OnlineShop.DAL.Repositories.Interfaces;
 
 namespace OnlineShop.BLL.Services.Implementations;
 
-public class UserService:IUserService
+public class UserService : IUserService
 {
-    
+    private readonly IUserRepository _userRepository;
+
+    public UserService(IUserRepository userRepository)
+    {
+        MapsterConfig.Configure();
+        
+        _userRepository = userRepository;
+    }
+
+    public async Task<UserResponseDTO> GetByIdAsync(string userId, CancellationToken cancellationToken = default)
+    {
+        var id = userId.Adapt<Guid>();
+
+        var user = await _userRepository.GetByIdAsync(id, cancellationToken);
+        if (user is null)
+        {
+            throw new EntityNotFoundException(nameof(User), id);
+        }
+
+        var userResponseDTO = user.Adapt<UserResponseDTO>();
+
+        return userResponseDTO;
+    }
+
+    public async Task LoginAsync(LoginRequestDTO loginRequestDTO, CancellationToken cancellationToken = default)
+    {
+        LoginValidator loginValidator = new LoginValidator();
+        await loginValidator.ValidateAndThrowAsync(loginRequestDTO, cancellationToken);
```

Comment:
And why do you need to create a validator in general, you can use auto-validation